on-start.sh

```diff
@@ -141,190 +173,246 @@ for host in ${peers[*]}; do
+            is_clone_process_run=1
+        fi
     fi
+    if [[ $is_clone_process_run == 0 ]]; then
+        # when clone approach not run then the member will try to join into the group
+        retry 120 ${mysql} -N -e "START GROUP_REPLICATION USER='repl', PASSWORD='password';"
+        log "INFO" "Group replication on (${cur_host}) has been taken place..."
+    fi
+}
 
-else
-    log "INFO" "A group is found and the primary host is '$primary_host'..."
-fi
-#####################################################################
-# End bootstrap process                                             #
-#####################################################################
-
-# Now start group replication inside the others
-#####################################################################
-# Begin joining others members to the group                         #
-#####################################################################
-declare -i host_idx=0
-
-for host in ${member_hosts[*]}; do
-
-    if [[ "$host" != "$primary_host" ]]; then
-        mysql="$mysql_header --host=$host"
-
-        # get the member state from performance_schema.replication_group_members
-        out=$(${mysql} -N -e "SELECT MEMBER_STATE FROM performance_schema.replication_group_members WHERE MEMBER_HOST = '$host';")
+# create mysql client with user exported in mysql_header and export password
+# this is to bypass the warning message for using password
+export mysql_header="mysql -u ${USER} --port=3306"
+export MYSQL_PWD=${PASSWORD}
+export member_hosts=$(echo -n ${hosts} | sed -e "s/,/ /g")
+export joining_for_first_time=0
+log "INFO" "Existing member in the group are: $member_hosts"
 
-        if [[ -z "$out" || "$out" == "OFFLINE" ]]; then
-            log "INFO" "Starting group replication on (${host})..."
+# wait for mysqld is getting ready for listen and serve
+wait_for_mysqld_running
 
-            ${mysql} -N -e "STOP GROUP_REPLICATION;"
+# ensure replication user for the group
+create_replication_user
 
-            # reset is needed for the first time creation
-            if [[ "${is_new[$host_idx]}" -eq "1" ]]; then
-                log "INFO" "RESET MASTER in host $host..."
-                ${mysql} -N -e "RESET MASTER;"
-            fi
+# ensure replication plugin is installed
+install_group_replication_plugin
 
-            ${mysql} -N -e "START GROUP_REPLICATION;"
+# ensure clone plugin is installed
+install_clone_plugin
 
-            log "INFO" "$host is joined the group $GROUP_NAME"
+# checking group replication existence
+check_existing_cluster "${member_hosts[*]}"
 
-        else
-            log "INFO" "Member state is '${out}' on host '${host}'..."
-        fi
-    fi
+# for cluster bootstrapping, by-default replica 0 will be selected for primary_host
+# it will also be overwritten in `wait_for_primary` function when it will get the updated primary_host
+export primary_host=$(get_host_name 0)
```

Comment:
We don't need this. It will be populated automatically in `wait_for_primary` function